torch/utils/data/datapipes/map/combining.py

```diff
@@ -0,0 +1,48 @@
+from torch.utils.data import MapDataPipe, functional_datapipe
+from typing import Optional, Sized, Tuple, TypeVar
+
+T_co = TypeVar('T_co', covariant=True)
+
+
+@functional_datapipe('concat')
+class ConcatMapDataPipe(MapDataPipe):
+    r""" :class:`ConcatMapDataPipe`.
+
+    Map DataPipe to concatenate multiple Map DataPipes.
```

Comment:
Can you also add comment to explain the actual index is based on the cumulative sum of source datapipes?